.changeset/tame-pianos-approve.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes an issue where the use of `Astro.rewrite` would trigger the invalid use of `Astro.request.headers`
```

Comment:
Are these two separate issues beign fixed? It's a little unclear.

Reply:
Yes. What should I reword?

Reply:
That's fine. Just wanted to confirm that they're separate.